### PR TITLE
Support for `share_properties` in `azurerm_storage_account`

### PR DESCRIPTION
**Pull request by yupwei68**
Fix https://github.com/terraform-providers/terraform-provider-azurerm/issues/7115 
https://github.com/terraform-providers/terraform-provider-azurerm/issues/12104

All tests have passed.

### CODE REVIEW

File: website/docs/r/storage_account.html.markdown [hunk 1]
```diff
@@ -399,6 +399,29 @@ The following attributes are exported in addition to the arguments listed above:
 
 ---
 
+A `share_properties` block supports the following:
+
+* `cors_rule` - (Optional) A `cors_rule` block as defined below.
+
+* `delete_retention_policy` - (Optional) A `delete_retention_policy` block as defined below.
```
Comment (katbyte) [inline website/docs/r/storage_account.html.markdown:406]: delete retention does not make sense - should this just be retention policy?

File: website/docs/r/storage_account.html.markdown [hunk 2]
```diff
@@ -399,6 +399,29 @@ The following attributes are exported in addition to the arguments listed above:
 
 ---
 
+A `share_properties` block supports the following:
```
Comment (katbyte) [inline website/docs/r/storage_account.html.markdown:402]: what type of share?

File: website/docs/r/storage_account.html.markdown [hunk 3]
```diff
@@ -399,6 +399,29 @@ The following attributes are exported in addition to the arguments listed above:
...
+* `smb` - (Optional) A `smb` block as defined below.
+
+
+---
+
+A `smb` block supports the following:
+
+* `versions` - (Optional) A set of SMB protocol versions. Possible values are `SMB2.1`, `SMB3.0`, and `SMB3.1.1`.
+
+* `authentication_methods` - (Optional) A set of SMB authentication methods. Possible values are `NTLMv2`, and `Kerberos`.
```
Comment (katbyte) [inline website/docs/r/storage_account.html.markdown:417]: should this be
```suggestion
* `authentication_types` - (Optional) A set of SMB authentication methods. Possible values are `NTLMv2`, and `Kerberos`.
```

File: website/docs/r/storage_account.html.markdown [hunk 4]
```diff
@@ -399,6 +399,29 @@ The following attributes are exported in addition to the arguments listed above:
...
+
+* `kerberos_ticket_encryption` - (Optional) A set of Kerberos ticket encryption. Possible values are `RC4-HMAC`, and `AES-256`.
```
Comment (katbyte) [inline website/docs/r/storage_account.html.markdown:419]:
```suggestion
* `kerberos_ticket_encryption_type` - (Optional) A set of Kerberos ticket encryption. Possible values are `RC4-HMAC`, and `AES-256`.
```

File: website/docs/r/storage_account.html.markdown [hunk 5]
```diff
@@ -399,6 +399,29 @@ The following attributes are exported in addition to the arguments listed above:
...
+
+* `channel_encryption` - (Optional) A set of SMB channel encryption. Possible values are `AES-128-CCM`, `AES-128-GCM`, and `AES-256-GCM`.
```
Comment (katbyte) [inline website/docs/r/storage_account.html.markdown:421]:
```suggestion
* `channel_encryption_type` - (Optional) A set of SMB channel encryption. Possible values are `AES-128-CCM`, `AES-128-GCM`, and `AES-256-GCM`.
```

**Review by katbyte — CHANGES_REQUESTED** [review]

File: website/docs/r/storage_account.html.markdown (hunk 2 above)
Comment (yupwei68) [reply website/docs/r/storage_account.html.markdown:402]: azurerm_storage_share

**Comment by yupwei68** [comment]
Thanks kt. All tests have passed. Please continue reviewing.

**Pull request merged**